django/db/models/base.py

```diff
@@ -1073,9 +1091,9 @@ def _save_table(
                 base_qs, using, pk_val, values, update_fields, forced_update
             )
             if force_update and not updated:
-                raise DatabaseError("Forced update did not affect any rows.")
+                raise cls.NotUpdated("Forced update did not affect any rows.")
             if update_fields and not updated:
-                raise DatabaseError("Save with update_fields did not affect any rows.")
+                raise cls.NotUpdated("Save with update_fields did not affect any rows.")
```

Comment:
Is there a reason not to use `self.NotUpdated` instead?

Reply:
yeah using `cls.NotUpdated` allows to differentiate the exact base that failed to be updated in MTI scenarios.

Reply:
thinking more about it might be better to always use `self.NotUpdated` as otherwise doing `except Child.NotUpdated` wouldn't be enough to call all exceptions.

Reply:
swapped it back to `self` 👍